app/src/main/java/org/schabi/newpipe/player/Player.java

```diff
@@ -4110,19 +4134,85 @@ public AppCompatActivity getParentActivity() {
+        isAudioOnly = !videoEnabled;
+        // When a user returns from background, controls could be hidden but SystemUI will be shown
+        // 100%. Hide it.
         if (!isAudioOnly && !isControlsVisible()) {
             hideSystemUIIfNeeded();
         }
+
+        // The current metadata may be null sometimes (for e.g. when using an unstable connection
+        // in livestreams) so we will be not able to execute the block below.
+        // Reload the play queue manager in this case, which is the behavior when the video stream
+        // played is not video only or when we don't know the index of the video renderer.
+        if (currentMetadata == null) {
+            reloadPlayQueueManager();
+            setRecovery();
+            return;
+        }
+
+        final int videoRenderIndex = getVideoRendererIndex();
+        final StreamInfo info = currentMetadata.getMetadata();
+
+        /* For video streams: we don't want to stream in background the video stream so if the
+        video stream played is not a video-only stream and if there is an audio stream available,
+        play this audio stream in background by reloading the play queue manager.
+        Otherwise the video renderer will be just disabled (because there is no
+        other stream for it to play the audio): if the video stream is video-only, only the audio
+        stream will be fetched and the video stream will be fetched again when the user return to a
+        video player.
+
+        For audio streams and audio live streams: nothing is done, it's not needed to reload the
+        player with the same audio stream.
+
+        For video live streams: the play queue manager is not reloaded if the stream source is a
+        live source (see VideoPlaybackResolver#resolve()) and if that's not the case, the
+        requirements for video streams is applied.
+
+        In the case where we don't know the index of the video renderer, the play queue manager
+        is also reloaded. */
+
+        final StreamType streamType = info.getStreamType();
+        final SourceType sourceType = videoResolver.getStreamSourceType()
+                .orElse(SourceType.VIDEO_WITH_SEPARATED_AUDIO);
+
+        final boolean isVideoWithSeparatedAudioOrVideoWithNoSeparatedAudioStreams =
+                sourceType == SourceType.VIDEO_WITH_SEPARATED_AUDIO
+                || (sourceType == SourceType.VIDEO_WITH_AUDIO_OR_AUDIO_ONLY
+                        && isNullOrEmpty(info.getAudioStreams()));
+        final boolean isVideoStreamTypeAndIsVideoOnlyStreamOrNoAudioStreamsAvailable =
+                streamType == StreamType.VIDEO_STREAM
+                        && isVideoWithSeparatedAudioOrVideoWithNoSeparatedAudioStreams
+                        || (streamType == StreamType.LIVE_STREAM
+                            && (sourceType == SourceType.LIVE_STREAM
+                                || isVideoWithSeparatedAudioOrVideoWithNoSeparatedAudioStreams));
```

Comment:
Idea: This code block is very hard to read (at least for me) and the variable is very long.
Maybe we could extract it into a Utility method and call that in L4196?

I think a method would also be nicer because we could do something like this:
```java
// If it's a video livestream -> do trackselector stuff
if (streamType == StreamType.LIVE_STREAM && sourceType == SourceType.LIVE_STREAM) {
   return true;
}

// If it's
// * a video with separated audio
// * OR a video with audio only THAT has no audio streams
// -> go on
if (sourceType == SourceType.VIDEO_WITH_SEPARATED_AUDIO
                || (sourceType == SourceType.VIDEO_WITH_AUDIO_OR_AUDIO_ONLY
                        && isNullOrEmpty(info.getAudioStreams())) {
   // If it's a video stream or a live stream -> do trackselector stuff
   if (streamType == StreamType.VIDEO_STREAM || streamType == StreamType.LIVE_STREAM) {
      return true;
   }
}

// default
return false
```